### PR TITLE
Cue filter scale input

### PR DESCRIPTION
**Pull request by LRahlff**
Added an input for scaling the input time

### CODE REVIEW

File: src/filters/filter_cue.cpp
Comment (Doralitze) [inline src/filters/filter_cue.cpp:1]: This file has grown to a complex state were I'm no longer capable of following what is going on simply based on the method names. Would it be possible to write some doc comments for every method?

Besides that I reviewed this file and the changes seam reasonable to me. Please elaborate on the comments where due.

**Review by Doralitze — CHANGES_REQUESTED** [review]
Please review the comments I attached. In general these changes look good but some minor things need to be worked on.

File: src/filters/filter_cue.cpp [hunk 1]
```diff
@@ -269,27 +270,33 @@ namespace dmxfish::filters {
 
     template<typename T>
     void filter_cue::calc_transition(double rel_time, transition_t transition, T start_value, T end_value, size_t ind) {
+
+        // offset is the value of the sigmoidal at point zero, so the offset of the x-axis
+        constexpr double offset = 0.0024726231566347743340599073722557624510930726498587540880605924;
+
```
Comment (LRahlff) [inline src/filters/filter_cue.cpp:276]: did you mean like this?

File: src/filters/filter_cue.cpp [hunk 2]
```diff
@@ -412,90 +414,95 @@ namespace dmxfish::filters {
...
             update_parameter_gui();
         }
-        if (frame < cues.at(active_cue).timestamps.size()) {
-            already_updated_last = false;
-            double rel_time =
-                    (*time - last_timestamp) / (cues.at(active_cue).timestamps.at(frame) - last_timestamp + start_time);
+        if (this->actual_values.frame < this->cues.at(this->actual_values.cue).timestamps.size()) {
+            this->last_values.updated = false;
+            double rel_time = (this->cues.at(this->actual_values.cue).timestamps.at(this->actual_values.frame) / this->time_scale - this->last_values.time_stamp + this->start_time) <= 0 ? 1 :
+                    (*this->time - this->last_values.time_stamp) / (this->cues.at(this->actual_values.cue).timestamps.at(this->actual_values.frame) / this->time_scale - this->last_values.time_stamp + this->start_time);
```
[the remark this answers is not in the document]
Comment (LRahlff) [reply src/filters/filter_cue.cpp]: now i introduced the current_time variable

File: src/filters/filter_cue.cpp (hunk 1 above)
Comment (Doralitze) [reply src/filters/filter_cue.cpp:276]: Yes.

File: src/filters/filter_cue.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (Doralitze) [reply src/filters/filter_cue.cpp]: Thanks.

**Pull request merged**